modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -924,8 +924,11 @@ private Set<String> _getAccountEntryRestrictedObjectRelationshipsNames(
 			}
 
 			com.liferay.object.model.ObjectDefinition objectDefinition2 =
-				_objectDefinitionLocalService.fetchObjectDefinition(
-					objectRelationship.getObjectDefinitionId2());
+				_objectDefinitionLocalService.
+					fetchObjectDefinitionByExternalReferenceCode(
+						objectRelationship.
+							getObjectDefinitionExternalReferenceCode2(),
+						contextCompany.getCompanyId());
```

Comment:
@evibudai can you change to fetch by `ID` first, and if `objectDefinition2` is `null` you fetch by `ERC`?

```
com.liferay.object.model.ObjectDefinition objectDefinition2 =
      _objectDefinitionLocalService.fetchObjectDefinition(
           objectRelationship.getObjectDefinitionId2());

if (objectDefinition2 == null) {
   objectDefinition2 = _objectDefinitionLocalService.
      fetchObjectDefinitionByExternalReferenceCode(....);
}

```

Reply:
Hi @carolmariaabb 
Sorry for the confusion, the exception happens actually inside 
`_objectDefinitionLocalService.fetchObjectDefinition();`	
I wrote NullPointerException, but it is an InvocationTargetException.

so we don't reach:
`if (objectDefinition2 == null) `

I have to change

			com.liferay.object.model.ObjectDefinition objectDefinition2 =
				_objectDefinitionLocalService.fetchObjectDefinition(
					objectRelationship.getObjectDefinitionId2());

for something else because of this failure

Reply:
/cc @marcelabc